From a6fd75bc44a9af1da9f528090b9fbba03065703e Mon Sep 17 00:00:00 2001
From: Simon Horman <horms+renesas@verge.net.au>
Date: Wed, 3 Jul 2013 13:38:18 +0900
Subject: [PATCH 0936/1083] Revert "drm/rcar-du: Add platform module device
 table"

This reverts commit d2a68e0355248cf1bfabffb9cb859cb1404e9d8a.
(cherry picked from commit 19a346a14066a9472521ea59bbbb91020b41bb6e)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 drivers/gpu/drm/rcar-du/rcar_du_drv.c | 13 -------------
 drivers/gpu/drm/rcar-du/rcar_du_drv.h | 15 ---------------
 2 files changed, 28 deletions(-)

diff --git a/drivers/gpu/drm/rcar-du/rcar_du_drv.c b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
index 69bd77a..26323bc 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_drv.c
+++ b/drivers/gpu/drm/rcar-du/rcar_du_drv.c
@@ -129,7 +129,6 @@ static int rcar_du_load(struct drm_device *dev, unsigned long flags)
 
 	rcdu->dev = &pdev->dev;
 	rcdu->pdata = pdata;
-	rcdu->info = (struct rcar_du_device_info *)pdev->id_entry->driver_data;
 	rcdu->ddev = dev;
 	dev->dev_private = rcdu;
 
@@ -314,17 +313,6 @@ static int rcar_du_remove(struct platform_device *pdev)
 	return 0;
 }
 
-static const struct rcar_du_device_info rcar_du_r8a7779_info = {
-	.features = 0,
-};
-
-static const struct platform_device_id rcar_du_id_table[] = {
-	{ "rcar-du-r8a7779", (kernel_ulong_t)&rcar_du_r8a7779_info },
-	{ }
-};
-
-MODULE_DEVICE_TABLE(platform, rcar_du_id_table);
-
 static struct platform_driver rcar_du_platform_driver = {
 	.probe		= rcar_du_probe,
 	.remove		= rcar_du_remove,
@@ -333,7 +321,6 @@ static struct platform_driver rcar_du_platform_driver = {
 		.name	= "rcar-du",
 		.pm	= &rcar_du_pm_ops,
 	},
-	.id_table	= rcar_du_id_table,
 };
 
 module_platform_driver(rcar_du_platform_driver);
diff --git a/drivers/gpu/drm/rcar-du/rcar_du_drv.h b/drivers/gpu/drm/rcar-du/rcar_du_drv.h
index ba8dd73..5c423dd 100644
--- a/drivers/gpu/drm/rcar-du/rcar_du_drv.h
+++ b/drivers/gpu/drm/rcar-du/rcar_du_drv.h
@@ -26,18 +26,9 @@ struct device;
 struct drm_device;
 struct drm_fbdev_cma;
 
-/*
- * struct rcar_du_device_info - DU model-specific information
- * @features: device features (RCAR_DU_FEATURE_*)
- */
-struct rcar_du_device_info {
-	unsigned int features;
-};
-
 struct rcar_du_device {
 	struct device *dev;
 	const struct rcar_du_platform_data *pdata;
-	const struct rcar_du_device_info *info;
 
 	void __iomem *mmio;
 	struct clk *clock;
@@ -61,12 +52,6 @@ struct rcar_du_device {
 	} planes;
 };
 
-static inline bool rcar_du_has(struct rcar_du_device *rcdu,
-			       unsigned int feature)
-{
-	return rcdu->info->features & feature;
-}
-
 int rcar_du_get(struct rcar_du_device *rcdu);
 void rcar_du_put(struct rcar_du_device *rcdu);
 
-- 
1.8.3.2

